spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -70,10 +72,9 @@
 	 * @throws IllegalArgumentException if the location URI is invalid
 	 */
 	public GoogleStorageResource(Storage storage, String locationUri, boolean autoCreateFiles) {
-		super(locationUri);
-
 		Assert.notNull(storage, "Storage object can not be null");
 		this.storage = storage;
+		this.location = new GoogleStorageLocation(locationUri);
```

Comment:
Can you just reuse the constructor below?

Reply:
Done.